extensions/amp-gist/0.1/amp-gist.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Embeds a Github gist
+ *
+ * Example:
+ * <code>
+ * <amp-gist
+ *   layout="responsive"
+ *   data-gistid="a19e811dcd7df10c4da0931641538497"
+ *   width="100"
+ *   height="100">
+ * </amp-gist>
+ * </code>
+ */
+
+import {getIframe} from '../../../src/3p-frame';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+
+export class AmpGist extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?HTMLIFrameElement} */
+    this.iframe_ = null;
+  }
+
+  /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://gist.github.com/', opt_onLayout);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
```

Comment:
We normally put new extensions behind an experiment flag until we are sure they are ready. We should do the same here: 
Please add:
- [ ] Entry in `experiments.js`
- [ ] Following guards in the code
```
const TAG = 'amp-gist';
```
```
  buildCallback() {
    user().assert(isExperimentOn(this.win, TAG),
        `Experiment "${TAG}" is disabled.`);
  }
```